src/components/Signup.js

```diff
@@ -1,18 +1,46 @@
-import React, {useRef} from 'react'
+import React, {useRef, useState} from 'react'
 // import {Form, Button, Card} from 'react-bootstrap'
-import { Form, Button, Card} from 'react-bootstrap'
+import { Form, Button, Card, Alert} from 'react-bootstrap'
+import { useAuth } from '../contexts/AuthContext'
+import { set } from 'firebase/database'
 
 export default function Signup() {
   const emailRef = useRef()
   const passwordRef = useRef()
   const passwordConfirmRef = useRef()
+  const {signup} = useAuth()
+  const [error, setError] = useState('')
+  const [loading, setLoading] = useState(false)
+
+  async function handleSubmit(e) {
+    //prevent the form from refreshing
+    e.preventDefault()   
+    //do the validation checks
+
+    if (passwordRef.current.value !==
+      passwordConfirmRef.current.value) {
+        return setError('Passwords do not match')
+      }
+    
+    try{
+      setError('')
+      // set up a load state, so when signing up the user, we diabled the "Sign Up" botton below, 
+      // so they don't automatically keep clicking the button and create multiple of accounts at the same time
+      setLoading(true)
+      await signup(emailRef.current.value, passwordRef.current.value)
+    } catch {
+      setError('Failed to create an account')
+    }
+    setLoading(false)
```

Comment:
your code will still work of course but this pattern makes things a bit nicer

```suggestion
    } finally {
      setLoading(false)
    }
```